src/models/transactions/common.ts

```diff
@@ -0,0 +1,136 @@
+import { ValidationError } from "../../common/errors"
+import { Memo, Signer } from "../common"
+
+const transactionTypes = [
+    "AccountSet",
+    "AccountDelete",
+    "CheckCancel",
+    "CheckCash",
+    "CheckCreate",
+    "DepositPreauth",
+    "EscrowCancel",
+    "EscrowCreate",
+    "EscrowFinish",
+    "OfferCancel",
+    "OfferCreate",
+    "Payment",
+    "PaymentChannelClaim",
+    "PaymentChannelCreate",
+    "PaymentChannelFund",
+    "SetRegularKey",
+    "SignerListSet",
+    "TicketCreate",
+    "TrustSet",
+]
+
+const isMemo = (obj: {Memo: Memo}): boolean => {
+    const memo = obj.Memo
+    const size = Object.keys(memo).length
+    const validData = memo.MemoData === undefined 
+        || typeof memo.MemoData === 'string'
+    const validFormat = memo.MemoFormat === undefined 
+        || typeof memo.MemoData === 'string'
+    const validType = memo.MemoType === undefined
+        || typeof memo.MemoType === 'string'
+
+    const otherKeys = Object.keys(memo).every((key: string) => {
+        return key === "MemoFormat" 
+            || key === "MemoData"
+            || key === "MemoType"
+    })
+
+    return (1 <= size && size <= 3) 
+        && validData 
+        && validFormat 
+        && validType
+        && otherKeys
+}
+
+const isSigner = (signer: Signer): boolean => {
+    return Object.keys(signer).length === 3
+        && typeof signer.Account === 'string'
+        && typeof signer.TxnSignature === 'string'
+        && typeof signer.SigningPubKey === 'string'
+}
+
+export interface CommonFields {
+  Account: string;
+  TransactionType: string;
+  Fee?: string;
+  Sequence?: number;
+  AccountTxnID?: string;
+  Flags?: number;
+  LastLedgerSequence?: number;
+  Memos?: Array<{ Memo: Memo }>;
+  Signers?: Array<Signer>;
+  SourceTag?: number;
+  SigningPubKey?: string;
+  TicketSequence?: number;
+  TxnSignature?: string;
+}
+
+/**
+ * verify the common fields of a transaction. The verify functionality will be
+ * optional, and will check transaction form at runtime. This should be called
+ * any time a transaction will be verified.
+ * 
+ * @param common - An interface w/ common transaction fields
+ * @returns - Void
+ * @throws - When the common param is malformed. 
+ */
+export function verifyCommonFields(common: CommonFields): void {
+    if (common.Account === undefined)
+        throw new ValidationError("CommonFields: missing field Account")
+    
+    if (typeof common.Account !== 'string')
+        throw new ValidationError("CommonFields: Account not string")
```

Comment:
Doesn't TypeScript handle this already? Or are we specifically doing runtime checks? This seems a bit excessive.

Reply:
I thought we were specifically doing runtime checks, similar to jsonschemas. If we're not checking the types, then we also shouldn't worry about any runtime verification IMO.